src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs

```diff
@@ -66,16 +65,49 @@ public class BlockchainServiceWeb3 : JsonRpcService
             var block = _blockManager.GetByHeight((ulong)blockNumber);
             if (block == null)
                 return null;
-            var txs = block!.TransactionHashes
+
+            ulong gasUsed = 0;
+            var txArray = new JArray();
+            if(block.TransactionHashes.Count>0)
+            {
+                var txs = block!.TransactionHashes
                 .Select(hash => _transactionManager.GetByHash(hash)!)
                 .ToList();
-            var gasUsed = txs.Aggregate<TransactionReceipt, ulong>(0, (current, tx) => current + tx.GasUsed);
-            var txArray = fullTx
-                ? Web3DataFormatUtils.Web3BlockTransactionArray(txs, block!.Hash, block!.Header.Index)
-                : new JArray();
+                gasUsed = txs.Aggregate<TransactionReceipt, ulong>(0, (current, tx) => current + tx.GasUsed);
+
+                if(fullTx) txArray = Web3DataFormatUtils.Web3BlockTransactionArray(txs, block!.Hash, block!.Header.Index);
+            }
+
+            return Web3DataFormatUtils.Web3Block(block!, gasUsed, txArray);
+        }
+
+        [JsonRpcMethod("testapi")]
+        private JObject? test(string blockRaw)
+        {
+            var block = Block.Parser.ParseFrom(blockRaw.HexToBytes());
+            Console.WriteLine(Web3DataFormatUtils.Web3Block(block,0,new JArray{}));
+            if (block == null)
+                return null;
+            Console.WriteLine("Tx count from block: "+  block.TransactionHashes.Count);
+            
+            ulong gasUsed = 0;
+            var txArray = new JArray{};
+            if(block.TransactionHashes.Count>0)
+            {
+                var txs = block!.TransactionHashes
+                .Select(hash => _transactionManager.GetByHash(hash)!)
+                .ToList();
+                Console.WriteLine("Entered here");
```

Comment:
Don't use Console.WriteLine please,  change it to Logger calls if you need this records. But this is debug out put,. it is better to remove it at all our logs are to big already.